ocaml/database/db_cache.ml

```diff
@@ -48,4 +53,11 @@ let apply_delta_to_cache entry db_ref =
   | Redo_log.WriteField (tblname, objref, fldname, newval) ->
       debug "Redoing write_field %s (%s) [%s -> %s]" tblname objref fldname
         newval ;
-      DB.write_field db_ref tblname objref fldname newval
+      let removed =
+        try lifecycle_state_of ~obj:tblname fldname = Removed_s
+        with Not_found ->
+          D.warn "no lifetime information about %s.%s, ignoring" tblname fldname ;
+          true
```

Comment:
WIll we ever remove 'removed' fields from the DB, or will they stay there forever with removed state? I think they have to stay, otherwise in the future we might reintroduce them (accidentally even) with different semantics.
OTOH the DB itself may not know about removed fields (just the datamodel it was generated from), and given that we access the datamodel here should we make it a hard failure if we ever find a truly removed field?

At least when tested by XenRT that sounds like it might be a useful thing to do.

Reply:
Yes, we should leave "removed" fields in the datamodel, even if it's just for documentation, but not the DB itself.

Hard failures in this code are risky for users, because it can lead to a loss of DB writes when using the redo-log for cases where it is meant to protect you from such loss.

Reply:
> Is removed = true the default, because Not_found implies that the field is not in the datamodel at all, so we can't write to it in any case?

That's the reasoning, yes